libraries/type-length-value/src/state.rs

```diff
@@ -936,23 +1231,132 @@ mod test {
         let mut state = TlvStateMut::unpack(&mut buffer).unwrap();
 
         // don't actually need to hold onto the data!
-        let _ = state.alloc::<TestVariableLen>(tlv_size).unwrap();
+        let _ = state.alloc_unique::<TestVariableLen>(tlv_size).unwrap();
         let test_variable_len = TestVariableLen {
             data: initial_data.to_string(),
         };
-        state.pack_variable_len_value(&test_variable_len).unwrap();
-        let deser = state.get_variable_len_value::<TestVariableLen>().unwrap();
+        state
+            .pack_variable_len_value_unique(&test_variable_len)
+            .unwrap();
+        let deser = state
+            .get_first_variable_len_value::<TestVariableLen>()
+            .unwrap();
         assert_eq!(deser, test_variable_len);
 
         // writing too much data fails
         let too_much_data = "This is a pretty cool test!?";
         assert_eq!(
             state
-                .pack_variable_len_value(&TestVariableLen {
+                .pack_variable_len_value_unique(&TestVariableLen {
                     data: too_much_data.to_string(),
                 })
                 .unwrap_err(),
             ProgramError::InvalidAccountData
         );
     }
+
+    #[test]
+    fn variable_len_value_allow_repeating() {
+        let initial_data = "This is a pretty cool test!";
+        // exactly the right size
+        let tlv_size = 8 + initial_data.len();
+        // Times two since we are going to pack two entries
+        let account_size = (get_base_len() + tlv_size) * 2;
+        let mut buffer = vec![0; account_size];
+        let mut state = TlvStateMut::unpack(&mut buffer).unwrap();
+
+        // Alloc and get entry number (first entry)
+        let (_, entry_number) = state
+            .alloc_allow_repeating::<TestVariableLen>(tlv_size)
+            .unwrap();
+        let test_variable_len = TestVariableLen {
+            data: initial_data.to_string(),
+        };
+
+        // Pack and get first
+        state
+            .pack_variable_len_value(&test_variable_len, entry_number)
+            .unwrap();
+        let deser = state
+            .get_first_variable_len_value::<TestVariableLen>()
+            .unwrap();
+        assert_eq!(deser, test_variable_len);
+
+        // Alloc and get entry number (second entry)
+        let (_, entry_number) = state
+            .alloc_allow_repeating::<TestVariableLen>(tlv_size)
+            .unwrap();
+        let test_variable_len = TestVariableLen {
+            data: initial_data.to_string(),
+        };
```

Comment:
To make these checks more useful, could you have different data for the second value? That way we're sure you're fetching and checking the second one